src/physics/em/detail/RBEnergySampler.i.hh

```diff
@@ -0,0 +1,66 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2021 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file RBEnergySampler.i.hh
+//---------------------------------------------------------------------------//
+#include <cmath>
+
+#include "base/Algorithms.hh"
+#include "random/distributions/BernoulliDistribution.hh"
+
+namespace celeritas
+{
+namespace detail
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Construct from incident particle and energy.
+ */
+CELER_FUNCTION
+RBEnergySampler::RBEnergySampler(const RelativisticBremNativeRef& shared,
+                                 const ParticleTrackView&         particle,
+                                 const CutoffView&                cutoffs,
+                                 const MaterialView&              material,
+                                 const ElementComponentId&        elcomp_id)
+    : shared_(shared)
+    , inc_energy_(particle.energy())
+    , gamma_cutoff_(cutoffs.energy(shared.ids.gamma))
+    , dxsec_(shared, particle, material, elcomp_id)
+{
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * Sample the bremsstrahlung photon energy based on G4eBremsstrahlungRelModel
+ * of the Geant4 10.7 release.
+ */
+template<class Engine>
+CELER_FUNCTION auto RBEnergySampler::operator()(Engine& rng) -> Energy
+{
+    // Min and max kinetic energy limits for sampling the secondary photon
+    Energy tmin = min(gamma_cutoff_, inc_energy_);
+    Energy tmax = min(shared_.high_energy_limit(), inc_energy_);
+
+    real_type density_corr = dxsec_.density_correction();
+    real_type xmin         = std::log(ipow<2>(tmin.value()) + density_corr);
+    real_type xrange = std::log(ipow<2>(tmax.value()) + density_corr) - xmin;
```

Comment:
```suggestion
    ReciprocalSampler sample_ee(ipow<2>(min_gamma_energy) + density_corr,
                             ipow<2>(inc_energy) + density_corr)
```
and below replace 
```c++
std::exp(xmin + generate_canonical(rng) * xrange)
```
with
```
sample_ee(rng);
```

This is basically the same (aside from the XS calculation) as is done in `SBEnergyDistHelper`, so maybe we can further reduce code duplication by templating that on the cross section code.

Reply:
Replaced with suggested changes for now.